src/js/modules/utilities/Logging.ts

```diff
@@ -9,40 +9,99 @@
  */
 
 import * as winston from "winston";
+import { Container, Logger, LoggerOptions } from "winston";
 
-export function newLogger(
-  module_name: string,
-  logLevel = "info",
-  simplify = false
-) {
-  const rpWasmFormatter = () => {
-    const logLineFormatter = winston.format.printf(
-      ({ level, message, label, timestamp }) => {
-        return `${timestamp} [${label}] ${level}: ${message}`;
-      }
-    );
-    const simpleLogLineFormatter = winston.format.printf(
-      ({ level, message, label }) => {
-        return `[${label}] ${level}: ${message}`;
-      }
-    );
+const maxLogFileSize = 5000000;
+
+const rpWasmFormatter = (moduleName: string, simplify: boolean) => {
+  const logLineFormatter = winston.format.printf(
+    ({ level, message, label, timestamp }) => {
+      return `${timestamp} [${label}] ${level}: ${message}`;
+    }
+  );
+  const simpleLogLineFormatter = winston.format.printf(
+    ({ level, message, label }) => {
+      return `[${label}] ${level}: ${message}`;
+    }
+  );
+
+  return (simplify ? [] : [winston.format.timestamp()]).concat([
+    winston.format.splat(),
+    winston.format.label({ label: moduleName }),
+    simplify ? simpleLogLineFormatter : logLineFormatter,
+  ]);
+};
+
+export class LogService {
+  private static instance: LogService;
+  private logFilePath: string;
+  private winstonContext: Container;
+
+  constructor() {
+    this.winstonContext = new winston.Container();
+  }
+
+  static getInstance(): LogService {
+    if (this.instance == undefined) {
+      this.instance = new LogService();
+      return this.instance;
+    }
+    return this.instance;
+  }
 
-    return (simplify ? [] : [winston.format.timestamp()]).concat([
-      winston.format.splat(),
-      winston.format.label({ label: module_name }),
-      simplify ? simpleLogLineFormatter : logLineFormatter,
-    ]);
-  };
-
-  const logger = winston.createLogger({
-    level: logLevel,
-    format: winston.format.combine(...rpWasmFormatter()),
-    defaultMeta: { service: module_name },
-    transports: [
-      new winston.transports.Console({
-        level: logLevel,
-      }),
-    ],
-  });
-  return logger;
+  setPath(path: string): void {
+    if (this.logFilePath == undefined) {
+      this.logFilePath = path;
+    }
+  }
+
+  getPath(): string {
+    return this.logFilePath;
+  }
+
+  createLogger(
+    moduleName: string,
+    option?: LoggerOptions,
+    simplify = false
+  ): Logger {
+    const loggerConfig = {
+      ...option,
+      level: option?.level,
+      format: winston.format.combine(...rpWasmFormatter(moduleName, simplify)),
+      defaultMeta: { service: moduleName },
+      transports: [
```

Comment:
What about the unhandled_exception handler? Why aren't we using that?

Reply:
No, because it doesn't take exception in an async code like `Promise` and all server implementation is into Promise.

https://github.com/vectorizedio/redpanda/blob/dev/src/js/modules/rpc/service.ts#L62

Reply:
Not sure i'm following. Are you trying to say we catch all exceptions anyway? What if the server throws ? Is it handled? service.ts#L69

Reply:
No, I forget to add the catch handler on those site, but I was trying to show you because the `unhandled_exception` work on our case